File: exercism/go/isogram/isogram.go
/*
 Package isogram offers a function for
 isogram detection.
*/
package isogram

import "unicode"

// IsIsogram determines if a word is an isogram.
func IsIsogram(word string) bool {
	cache := map[rune]bool{}
	for _, char := range word {
		if char == ' ' || char == '-' {
			continue
		}
		letter := unicode.ToLower(char)
		if _, ok := cache[letter]; ok {
			return false
		}
		cache[letter] = true
	}
	return true
}

File: exercism/c/isogram/src/isogram.c
#include <ctype.h>
#include <stdio.h>

#include "isogram.h"

bool is_isogram(const char phrase[]) {
  if (phrase == NULL) {
    return false;
  }

  bool seenLetters[26] = {false};

  for (size_t i = 0; phrase[i] != '\0'; i++) {
    unsigned char c = phrase[i];
    char letter = tolower(c);
    if (!isalpha(letter)) {
      continue;
    }
    if (seenLetters[letter - 'a']) {
      return false;
    }
    seenLetters[letter - 'a'] = true;
  }
  return true;
}
